### ISSUE TITLE
Incorrect ShadowedImports warning on relative '.' import

### ISSUE THREAD

**Issue by rvanlaar** [microsoft/pylance-release#3725]
## Environment data

-   Language Server version: v2022.11.40
-   OS and version: Ubuntu 22.04
-   Python version: 3.10.6

## Code Snippet

In a python package called `my_app`:
`types.py`
```

from typing import Literal

FileTypes = Literal["pdf", "html"]
```

`failure.py`
```

from .types import FileTypes
```
`works.py`
```
from my_app.types import FileTypes
```

## Expected behavior

Expected behavior is no warnings

## Actual behavior

When using a relative import with `.` notation Pylance assumes that the real `types` module is being shadowed.
It does understand it when the import specifies the full path.

The message:
"/path/to/my_app/types.py" is overriding the stdlib module "types"PylancereportShadowedImports

**Comment by rchiodo** [comment]
Thanks for the bug. This was actually reported in pyright (and fixed) here:
https://github.com/microsoft/pyright/issues/4269

This should be fixed in our next release.

**Comment by rvanlaar** [comment]
Thank you for the kind and apt reply.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2022.12.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20221221-7-December-2022-PreRelease)
